drivers/shared/hostnames/mount.go

```diff
@@ -0,0 +1,75 @@
+package hostnames
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/nomad/plugins/drivers"
+)
+
+// GenerateEtcHostsMount writes a /etc/hosts file using the network spec's
+// hosts configuration, and returns a mount config so that task drivers can
+// bind-mount it into the resulting task's filesystem. The extraHosts
+// parameter is expected to be the same format as the extra_hosts field from
+// the Docker or containerd drivers: []string{"<hostname>:<ip address>"}
+func GenerateEtcHostsMount(taskDir string, conf *drivers.NetworkIsolationSpec, extraHosts []string) (*drivers.MountConfig, error) {
+	if conf == nil || conf.Mode != drivers.NetIsolationModeGroup {
+		return nil, nil
+	}
+	hostsCfg := conf.HostsConfig
+	if hostsCfg == nil || hostsCfg.Address == "" || hostsCfg.Hostname == "" {
+		return nil, nil
+	}
+
+	content := fmt.Sprintf(`# this file was generated by Nomad
+127.0.0.1 localhost
+::1 localhost
+::1 ip6-localhost ip6-loopback
+fe00::0 ip6-localnet
+ff00::0 ip6-mcastprefix
+ff02::1 ip6-allnodes
+ff02::2 ip6-allrouters
+ff02::3 ip6-allhosts
+
+# this entry is the IP address and hostname of the allocation
+# shared with tasks in the task group's network
+%s %s
+`, hostsCfg.Address, hostsCfg.Hostname)
+
+	if len(extraHosts) > 0 {
+		content += "\n# these entries are extra hosts added by the task config"
```

Comment:
My past made me cringe whenever I see `+=` string concatenation. [`strings.Builder`](https://golang.org/pkg/strings/#Builder) might be handy. It's not a big deal, as I hope the file is going to be small, and I don't know if golang is smarter under the hood about this.

Reply:
Fair enough, will fix.

Reply:
Fixed in 40b1545